debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -691,6 +691,9 @@ protected void handleQueryEvent(Event event) throws InterruptedException {
             // This is an XA transaction, and we currently ignore these and do nothing ...
             return;
         }
+        if (sql.toUpperCase().startsWith("INSERT ") || sql.toUpperCase().startsWith("UPDATE ") || sql.toUpperCase().startsWith("DELETE")) {
```

Comment:
That repeated conversion in the hot path was bugging me :) I've pushed a commit which avoids it altogether.

Reply:
I knew you'll do it :-)